docs/pages/concepts/offloading/namespace-replication.md

```diff
@@ -0,0 +1,83 @@
+  It is worth noting that the NamespaceMap status represent the source of truth to know the status of replicated namesapces on foreign clusters.
+
+#### Controllers
+
+The namespace replication logic is composed of several distinct controllers, responsible for different aspects of namespace controllers:
+
+* **NamespaceOffloading Controller**: processes the NamespaceOffloading spec and outputs namespace entries to NamespaceMap specs.
+* **NamespaceMap Controller**: enforces the namespace creation of the remote namespaces on remote clusters and updates the status of the NamespaceMaps.
+* **OffloadingStatus Controller**: updates the status of the NamespaceOffloading resources by reading the NamespaceMaps statuses.
+
+#### Workflow
+
+In the next figure, you can observe a representation of the overall workflow steps of the replication process:
+
+![](/images/namespace-replication/replication.png)
+
+We can resume the steps in:
+
+1. When the user creates/updates a **NamespaceOffloading** object in a Liqo-enabled namespace **(Step 1 in the figure)**, the Liqo logic processes the resource spec.
+
+2. After having detected the virtual-nodes compliant with selector, a namespace creation request is entered in every NamespaceMap associated with a selected cluster **(Step 2 in the figure)**.
+In particular, it considers the **ClusterSelector** field to select on which cluster the current namespace should be replicated.
+
+More precisely, the controller responsible for this reconciliation is the **NamespaceOffloading Controller**. 
+It processes the *NamespaceOffloading* spec fields, by inserting the namespace creation requests in the *spec.DesiredMapping* field of **NamespaceMap** instances of selected clusters.
+The request format is an entry consisting of the local namespace name as a key and the name of the remote namespace as a value.
+In addition, this operation has to be performed every time a new virtual-node joins the cluster after a new peering is established.
+
+3. Once a *NamespaceMap* is updated, it is time to create the corresponding remote namespace. 
+The *NamespaceMap Controller* reconciles the NamespaceMap resources, creating the remote namespaces and storing the operation results in the corresponding NamespaceMap.
+More precisely, the occurred creation is saved in the *status.CurrentMapping* field, as you can see in **Step 3 of the figure**.
+The result format is similar to request one: the key is the name of the local namespace, while the value is composed of the remote name and the actual remote namespace phase.
+
+In addition to the actual creation of namespaces, the *NamespaceMap Controller* (1) periodically checks that each entry in the *spec.desiredMapping* field has an associated remote namespace and (2) performs health checks on the remote namespaces. The result of all its operations are stored in the *status.CurrentMapping* field in the NamespaceMap resource.
```

Comment:
Is this step (4)?
Otherwise, step (4) is missing.

Reply:
We added it, thank you.